### DISCUSSION TITLE
how to convert Int64 NDArray to float32 NDArray?

### DISCUSSION THREAD

**Discussion by kameronren** [deepjavalibrary/djl#1032] [Q&A] [answered]
how to convert Int64 NDArray to float32 NDArray?

`var cls_pred = cls_preds.get(0).argMax(2);
 var cls_label = cls_labels.get(0);`

**cls_pred** is a Int64 NDArray; **cls_label** is a float32 NDArray. Now I want to get the **'eq'** function results, but a Int64 NDArray can not  to compare with a float32 NDArray。

**Comment by zachgk** [comment, marked as answer]
You can use `array.toType` ([Javadoc](https://javadoc.io/static/ai.djl/api/0.11.0/ai/djl/ndarray/NDArray.html#toType-ai.djl.ndarray.types.DataType-boolean-)).

**Reply by kameronren** [reply]
thanks. your method is really works.
